lib/rexml/parsers/baseparser.rb

```diff
@@ -461,9 +459,9 @@ def pull_event
                 else
                   raise REXML::ParseException.new("Malformed CDATA: Missing end ']]>'", @source)
                 end
+              else
+                raise REXML::ParseException.new("Malformed node", @source)
```

Comment:
How about more information why this is a malformed node something like the following?

```suggestion
                raise REXML::ParseException.new("Malformed node: Started with '<!' but not a comment nor CDATA", @source)
```